src/inversify.config.ts

```diff
@@ -31,18 +32,28 @@ export function initContainer(): Container {
   container.bind<UriFactory>(UriFactory).toConstantValue(new UriFactory())
 
   // Session
-  const config: CookieConfig = {
+  const cookieConfig: CookieConfig = {
     cookieName: getEnvOrThrow('COOKIE_NAME'),
-    cookieSecret: getEnvOrThrow('COOKIE_SECRET')
+    cookieSecret: getEnvOrThrow('COOKIE_SECRET'),
+    cookieDomain: getEnvOrThrow('COOKIE_DOMAIN')
   }
   const sessionStore = new SessionStore(new IORedis(`${getEnvOrThrow('CACHE_SERVER')}`))
   container.bind(SessionStore).toConstantValue(sessionStore)
-  container.bind(TYPES.SessionMiddleware).toConstantValue(SessionMiddleware(config, sessionStore))
+  container.bind(TYPES.SessionMiddleware).toConstantValue(SessionMiddleware(cookieConfig, sessionStore))
   container.bind(TYPES.AuthMiddleware).toConstantValue(AuthMiddleware(
     getEnvOrThrow('CHS_URL'),
     new UriFactory(),
     commonAuthMiddleware
   ))
+  const authConfig: AuthConfig = {
+    accountUrl: getEnvOrThrow('ACCOUNT_URL'),
+    accountRequestKey: getEnvOrThrow('ACCOUNT_OAUTH2_REQUEST_KEY'),
+    accountClientId: getEnvOrThrow('ACCOUNT_CH_CLIENT_ID'),
+    chsUrl: getEnvOrThrow('CHS_URL'),
+  }
+  container.bind(TYPES.CompanyAuthMiddleware).toConstantValue(
+    CompanyAuthMiddleware(cookieConfig, authConfig, createLogger(APP_NAME)))
```

Comment:
No need to recreate a logger, just reuse the one defined here https://github.com/companieshouse/dissolution-web/pull/23/files#diff-06d803e02ba0c13e85350152879527f5R31